client/packages/system/src/InventoryAdjustmentReason/Components/InventoryAdjustmentReasonSearchInput.tsx

```diff
@@ -13,34 +13,33 @@ import {
   InventoryAdjustmentReasonRowFragment,
 } from '../api';
 
+export enum Adjustment {
+  Reduction,
+  Addition,
+  None,
+}
+
 interface InventoryAdjustmentReasonSearchInputProps {
   value: InventoryAdjustmentReasonRowFragment | null;
   width?: number | string;
   onChange: (
     inventoryAdjustmentReason: InventoryAdjustmentReasonRowFragment | null
   ) => void;
   autoFocus?: boolean;
-  stockReduction: number;
+  adjustment: Adjustment;
   isError?: boolean;
 }
 
 export const InventoryAdjustmentReasonSearchInput: FC<
   InventoryAdjustmentReasonSearchInputProps
-> = ({
-  value,
-  width,
-  onChange,
-  autoFocus = false,
-  stockReduction,
-  isError,
-}) => {
+> = ({ value, width, onChange, autoFocus = false, adjustment, isError }) => {
   const { data, isLoading } =
     useInventoryAdjustmentReason.document.listAllActive();
-  const disabled = data?.totalCount === 0 || stockReduction === 0;
-  const isRequired = !disabled && stockReduction !== 0;
+  const isRequired = data?.totalCount !== 0;
+  const isDisable = adjustment === Adjustment.None || !isRequired;
```

Comment:
I reshuffled this, using enum actually identified a problem, typescript complained since, i.e. [example](https://www.typescriptlang.org/play?#code/KYOwrgtgBAggzgUXNA3gKCpqAtYAnAewBoMsYAjAgN2F0JKygCFgAbAgdzoLQF800AYwIg4AFyiCAFsEEBrKAF4oACmEgAJgEsxWkQC4olAq2ABDEEShUzrMMEPJy+AJRKAfFHRZ14qADMtPD9ldW1dESgAHyjrW3slRWUABgBuUkkRPzhZEQ0lKABCQOCJADIyuLtgIqSoNIEsPGAxMDwQLxLxKxyw-n4hLIlpWTkAdR0pJEgCtTydPRBDY1MLKxtqw3hpiDdFT29MXwkuiVD5iI6YqoSk5W3kADpudJ8hqF68guKgvwqbmqFOoPSDPfAEdIZZqtdqdX5iHq5TT9AS+EzAR7sADmahk8hUYjw9isyRcLkGonRmIIOJG+MJxKgAEYyQIgA)

Reply:
isDisabled